pkg/backingstore/backingstore.go

```diff
@@ -420,7 +422,18 @@ func RunDelete(cmd *cobra.Command, args []string) {
 	backStore.Namespace = options.Namespace
 	backStore.Spec = nbv1.BackingStoreSpec{}
 
-	if !util.KubeDelete(backStore) {
+	nbClient := system.GetNBClient()
+
+	poolinfo, err := nbClient.ReadPoolAPI(nb.ReadPoolParams{Name: backStore.Name})
+	if err != nil {
+		log.Fatalf("❌ Failed to read BackingStore info: %s", err)
+	} else if poolinfo.Undeletable == "IN_USE" {
+		log.Fatalf(`❌ Could not delete BackingStore %q in namespace %q as it is being used by one or more buckets`,
+			backStore.Name, backStore.Namespace)
+	} else if poolinfo.Undeletable == "DEFAULT_RESOURCE" {
+		log.Fatalf(`❌ Could not delete BackingStore %q in namespace %q as it is the default resource of one or more accounts`,
+			backStore.Name, backStore.Namespace)
+	} else if !util.KubeDelete(backStore) {
```

Comment:
I would avoid `else if` here - it's hard to follow the flow when the delete call is hidden inside a compound if statement.

Reply:
changed to switch